curated_transformers/cli/debug_pieces.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, Dict, List, Optional, cast
+import numpy
+from pathlib import Path
+from spacy import util
+from spacy.cli._util import (
+    debug_cli,
+    import_code,
+    parse_config_overrides,
+    show_validation_error,
+)
+from spacy.schemas import ConfigSchemaTraining
+from spacy.tokens import Doc
+from spacy.util import registry, resolve_dot_names
+from typer import Argument as Arg, Context, Option as Opt
```

Comment:
Note that we'll be [moving](https://github.com/explosion/spaCy/pull/12258) from `typer` to `radicli` for `v4` so maybe it makes sense already using that here now?

Reply:
But this needs to be compatible with spaCy 3.6, would combining these work? We plan to make a `v4` branch of curated transformers for distillation. Maybe we can change it there later?

Reply:
It probably wouldn't work if spaCy 3.6 were used with an older Python version, so yea, let's keep this for the `v4` branch.